### PR TITLE
Use python to define the E2E test workflow for kfctl.

### PR DESCRIPTION
**Pull request by jlewi**
* We need to write an E2E test for kfctl upgrade (kubeflow/kfctl#35).

  Before we do that we want to remove ksonnet from our existing E2E test.

  The E2E test for upgrades will be very similar to the kfctl E2E test
  so it makes sense to convert that test to python before writing
  the upgrade test.

* Related to kubeflow/kubeflow#3035 migrate e2e tests off ksonnet

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/4148)

### CODE REVIEW

**Comment by jlewi** [comment]
So close.
Workflow ran and most steps completed successfully
http://testing-argo.kubeflow.org/workflows/kubeflow-test-infra/kubeflow-presubmit-kfctl-go-iap-istio-4148-96825a4-7760-946c?tab=workflow

It looks like delete test-dir failed.

No longs show up if we filter by pod label (probably because of the known stackdriver issue).
But if we get logs for the pod id it works.

The error is

```
rm: cannot remove '/mnt/test-data-volume/kubeflow-presubmit-kfctl-go-iap-istio-4148-96825a4-7760-946c/src/kubeflow/kubeflow/output/artifacts/logs/.nfs000000000070af1c0000003b': Device or resource busy
```

**Comment by jlewi** [comment]
Looks like both basic-auth and iap workflows ran and only the step test_dir delete failed.

**Comment by jlewi** [comment]
Tests passed
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/kubeflow_kubeflow/4148/kubeflow-presubmit/1176949535604215811/

**Comment by jlewi** [comment]
/assign @gabrielwen 
/assign @yanniszark 

@yanniszark @gabrielwen This is ready for review. The tests passed before I rebased. The current run was triggered by the rebase but I expect them to pass as well.

File: py/kubeflow/kubeflow/ci/kfctl_e2e_workflow.py [hunk 1]
```diff
@@ -0,0 +1,641 @@
...
+    #*************************************************************************
+    # Test katib deploy
+    step_name = "test-katib-deploy"
+    command = ["python",
+               "-m",
+               "testing.test_deploy",
+               "--project=kubeflow-ci",
+               # TODO(jlewi): Do we need a GITHUB_TOKEN? I'm guessing that
+               # was for ksonnet.
+               # "--github_token=$(GITHUB_TOKEN)",
```
Comment (gabrielwen) [inline py/kubeflow/kubeflow/ci/kfctl_e2e_workflow.py]: seems like we can remove this?

File: py/kubeflow/kubeflow/ci/kfctl_e2e_workflow.py [hunk 2]
```diff
@@ -0,0 +1,641 @@
...
+
+    #*************************************************************************
+    # Test pytorch job
+    step_name = "pytorch-job-deploy"
+    command = [ "python",
+                "-m",
+                "testing.test_deploy",
+                "--project=kubeflow-ci",
+                # TODO(jlewi): Do we still need a GITHUB_TOKEN?
+                # "--github_token=$(GITHUB_TOKEN)",
```
Comment (gabrielwen) [inline py/kubeflow/kubeflow/ci/kfctl_e2e_workflow.py]: ditto.

**Comment by jlewi** [comment]
@yanniszark Did you want to take a look at this?

**Comment by jlewi** [comment]
Looks like the workflow 
	http://testing-argo.kubeflow.org/workflows/kubeflow-test-infra/kubeflow-presubmit-jupyterui-release-4148-1909805-8608-4f44?tab=workflow

Failed. I'm not sure why we are even running that workflow got triggered. I suspect an issue with computing the modified files because of the clone depth.

**Comment by jlewi** [comment]
```
INFO|2019-09-27T09:38:16|/src/kubeflow/testing/py/kubeflow/testing/util.py|45| Running: git merge-base HEAD remotes/origin/master 
cwd=/src/kubeflow/kubeflow
INFO|2019-09-27T09:38:16|/src/kubeflow/testing/py/kubeflow/testing/util.py|60| Subprocess output:

WARNING|2019-09-27T09:38:17|/src/kubeflow/testing/py/kubeflow/testing/run_e2e_workflow.py|164| git merge-base failed; see https://github.com/kubeflow/kubeflow/issues/3523. Diff will be computed against the current master and therefore files not changed in the PR might be considered when determining which tests to trigger
...
INFO|2019-09-27T09:39:30|/src/kubeflow/testing/py/kubeflow/testing/run_e2e_workflow.py|239| Triggering workflow jupyterui-release because components/jupyter-web-app/backend/kubeflow_jupyter/common/api.py in dir components/jupyter-web-app/* is modified.
```

**Pull request merged**